homeassistant/components/verisure/__init__.py

```diff
@@ -93,16 +95,47 @@ def setup(hass, config):
     ):
         discovery.load_platform(hass, component, DOMAIN, {}, config)
 
+    @verify_domain_control(hass, DOMAIN)
     def capture_smartcam(service):
         """Capture a new picture from a smartcam."""
         device_id = service.data.get(ATTR_DEVICE_SERIAL)
-        HUB.smartcam_capture(device_id)
-        _LOGGER.debug("Capturing new image from %s", ATTR_DEVICE_SERIAL)
+        try:
+            HUB.smartcam_capture(device_id)
```

Comment:
So the errors are because it can only decorate async functions. Forgot to mention that. Easy to fix.

Change `def capture_smartcam` -> async def capture_smartcam`

And then

```suggestion
            await hass.async_add_executor_job(HUB.smartcam_capture, device_id)
```

Reply:
Thanks